### ISSUE TITLE
Custom Input  do not receive record when inside ArrayInput 

### ISSUE THREAD

**Issue by ThieryMichel** [marmelab/react-admin#1783]
**What you were expecting:**
I tried to add a custom Input inside an ArrayInput.

**What happened instead:**
My custom input received an undefined record props.

**Steps to reproduce:**
```js
const CustomInput = ({ record, source }) => <span>record is undefined here</span>;

<SimpleForm>
        <ArrayInput source="list" >
            <SimpleFormIterator>
                <CustomInput source="data" />
            </SimpleFormIterator>
        </ArrayInput>
    </SimpleForm>
```

**Environment**

* React-admin version: 2.0.0-RC1
* React version: 16.3

**Comment by jpetitcolas** [comment]
As far as I understand how it behaves, it is not expected to get the `record` at this point. Instead, it relies on `redux-form` state. `react-admin` handles the inputs using the `form.record-form.${source}` state property.

Your issue may be related to https://github.com/marmelab/react-admin/issues/1788, mighn't it?

**Comment by Luwangel** [comment]
Fixed in PR #1792
